cohortextractor/tpp_backend.py

```diff
@@ -316,6 +324,12 @@ def get_query(self, column_name, query_type, query_args):
         self._current_column_name = column_name
         return_value = method(**query_args)
         self._current_column_name = None
+        # We want to support returning a (columns, sql, setup_queries) triple
+        # from query methods but without, for now, having to rewrite all the
+        # old methods which just return (columns, sql). So we catch old-style
+        # return values here and add an empty `setup_queries` list to the end.
+        if isinstance(return_value, tuple) and len(return_value) == 2:
```

Comment:
I'm thinking about the refactoring we know we want to do, but not just yet, and how we signal that for future us.  In particular it would be good for legacy callers to be marked as such, somehow. We could consider a deprecation warning approach, but that means we'd also want to incorporate some kind of checks to surface them; or we could add comments in all the callers; or make a ticket (problem: we'll probably forget it exists). Any thoughts?

Reply:
In this case, it's not that there's a new shiny function signature and a deprecated one and we want to move everything over to the new one. Rather, both are a fudge at the moment and I think the right refactor involves a bigger change.

I'll write up what I think needs doing in a ticket. We might well forget about it, but it's better than nothing.